addons/website/models/website_menu.py

```diff
@@ -147,12 +164,16 @@ def is_active(self):
         if not menu_url:
             return False
 
-        menu_url = unslug_url(menu_url)
-        request_url = unslug_url(request.httprequest.path)
-        if request_url == menu_url:
+        menu_url = url_parse(unslug_url(menu_url))
+        request_url = url_parse(unslug_url(request.httprequest.url))
+        if menu_url.path == request_url.path and menu_url.decode_query() == request_url.decode_query():
+            if menu_url.netloc and menu_url.netloc != request_url.netloc:
+                # correct path but not correct domain
+                return False
```

Comment:
- Is it necessary to nest this if in the other one? If yes, maybe a comment because I don't understand why.
- Should we actually return False in that case? If fact, even in the case "not menu_url" above, should not we let it go though up to the children comparison as currently an URL on a menu with children does not make sense? :grimacing:

Reply:
For first point, do you ask if I can merge both if and let this case fallback on the final return of the method?
```py
# [...]
if abc:
    if xyz:
        return False
    return True
# [...]
return False
```
-> 
```py
# [...]
if abc and not xyz:
    return True
# [...]
return False
```
Yes it is possible but I don't like that because it's a specific corner case, I like it encapsulated in its if with a comment in the main if, it makes it easier to read and less prone to a later bug where one would add a condition below before the final return that would make this case go through it too

> maybe a comment because I don't understand why.

There is one: `# correct path but not correct domain`, let me know what should I make more clear, right now it's quite "simple": if the path and the query string match, return true except if the domain don't match (which is the corner case)

For second point, there was indeed a case not handled: we don't compare the menu url **at all** if it has submenus. I fixed that and added quite some tests to cover everything